InvenTree/templates/js/translated/forms.js

```diff
@@ -1538,11 +1538,18 @@ function addSecondaryModal(field, fields, options) {
         var url = secondary.api_url || field.api_url;
 
         // If the "fields" attribute is a function, call it with data
-        if (secondary.fields instanceof Function) {
+        if (secondary.fields instanceof Function || secondary.fieldsFunction instanceof Function) {
 
             // Extract form values at time of button press
             var data = extractFormData(fields, options);
 
+            // Backup and execute fields function in sequential executions of modal
```

Comment:
This allows the value of the next valid shipment reference to be recalculated each time the secondary button is pressed while the primary modal is open.

Further tests may be necessary to ensure this doesn't conflict with other secondary modals.